parachains/common/src/xcm_config.rs

```diff
@@ -63,3 +63,24 @@ impl<Location: Get<MultiLocation>> ContainsPair<MultiAsset, MultiLocation>
 		matches!(asset.id, Concrete(ref id) if id == origin && origin == &Location::get())
 	}
 }
+
+pub struct RelayOrOtherSystemParachains<
+	SystemParachainMatcher: Contains<MultiLocation>,
+	Runtime: parachain_info::Config,
+> {
+	_runtime: PhantomData<(SystemParachainMatcher, Runtime)>,
+}
+impl<SystemParachainMatcher: Contains<MultiLocation>, Runtime: parachain_info::Config>
+	Contains<MultiLocation> for RelayOrOtherSystemParachains<SystemParachainMatcher, Runtime>
+{
+	fn contains(l: &MultiLocation) -> bool {
+		let self_para_id: u32 = parachain_info::Pallet::<Runtime>::get().into();
+		if let MultiLocation { parents: 0, interior: X1(Parachain(para_id)) } = l {
+			if *para_id == self_para_id {
+				return false
```

Comment:
here should be probably `return true`, because we want to `waive` runtime itself not to pay fees (used for `XcmFeesToAccount`), 
with `return false` runtime itself wont be waived, means it needs to pay

other question is if matching of `MultiLocation { parents: 0, interior: X1(Parachain(para_id))` shouldnt be replaced with `Here` or better just remove this if for self-detection

Reply:
I think we avoided matching ourself for the asset-conversion because we started trying to asset transfer native assets - we have explicit checks in the asset-coversion pallet now so I don't think this exception is now needed.